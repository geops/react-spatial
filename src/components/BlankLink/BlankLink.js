import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * Link to open in a new window.
   */
  href: PropTypes.string.isRequired,
  /**
   * Children content of the link.
   */
  children: PropTypes.node.isRequired,
  /**
   * Title attribute.
   */
  title: PropTypes.string,
  /**
   * CSS class.
   */
  className: PropTypes.string,
};

const defaultProps = {
  className: undefined,
  title: undefined,
};

/**
 * This component displays a simple a HTML element which opens the link provided
 * in a blank window.
 */
class BlankLink extends PureComponent {
  render() {
    const { href, children, title, className } = this.props;
    return (
      <a
        href={href}
        title={title}
        className={className}
        target="_blank"
        rel="noopener noreferrer"
        tabIndex="0"
      >
        {children}
      </a>
    );
  }
}

BlankLink.propTypes = propTypes;
BlankLink.defaultProps = defaultProps;

export default BlankLink;
