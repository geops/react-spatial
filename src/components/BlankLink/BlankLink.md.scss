.tm-blank-link-example {
  @include a();

  a {
    display: block;
    padding: 10px;
  }
}
