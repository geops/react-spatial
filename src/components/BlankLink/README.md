#

This demonstrates the use of BlankLink.

```jsx
import React from  'react'
import BlankLink from 'react-spatial/components/BlankLink';

<div class="tm-blank-link-example">
  <BlankLink href="https://geops.de">Visit the geOps website!</BlankLink>
</div>;
```
