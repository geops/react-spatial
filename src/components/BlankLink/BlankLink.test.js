import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BlankLink from './BlankLink';

configure({ adapter: new Adapter() });

describe('BlankLink', () => {
  describe('when no properties are set', () => {
    let spy = null;

    beforeEach(() => {
      window.console.error = jest.fn().mockImplementation(() => {});
      spy = jest.spyOn(window.console, 'error');
    });

    afterEach(() => {
      spy.mockRestore();
      window.console.error.mockRestore();
    });

    test('displays 2 errors in the console', () => {
      shallow(<BlankLink />);
      expect(spy).toHaveBeenCalledTimes(2);
    });

    test('matches snapshot', () => {
      const component = renderer.create(<BlankLink />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when properties are set', () => {
    const href = 'http://google.com';
    const label = 'foo';
    const title = 'bar';
    const className = 'qux';

    test('matches snapshot', () => {
      const component = renderer.create(
        <BlankLink href={href} title={title} className={className}>
          {label}
        </BlankLink>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
