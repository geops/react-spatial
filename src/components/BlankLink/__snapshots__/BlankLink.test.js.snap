// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankLink when no properties are set matches snapshot 1`] = `
<a
  rel="noopener noreferrer"
  tabIndex="0"
  target="_blank"
/>
`;

exports[`BlankLink when properties are set matches snapshot 1`] = `
<a
  className="qux"
  href="http://google.com"
  rel="noopener noreferrer"
  tabIndex="0"
  target="_blank"
  title="bar"
>
  foo
</a>
`;
